### PR TITLE
fix: add max pagesize to http endpoints

### PR DESCRIPTION
**Pull request by zachterrell57**
## Motivation

https://github.com/farcasterxyz/hub-monorepo/issues/1715

## Change Summary

Add a simple MAX_SIZE to the pagination function

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [x] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [x] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

---

## PR-Codex overview
The focus of this PR is to set a global maximum page size for all endpoints in the application.

### Detailed summary
- Added a global constant `MAX_PAGE_SIZE` with a value of 1000
- Updated the `getPageOptions` function to limit the page size to the global maximum
- Ensured that the page size does not exceed the global maximum limit

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by sds — APPROVED** [review]
Thanks! Seems reasonable to me. Will tag others for sign off.

File: apps/hubble/src/rpc/httpServer.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ type StaticEncodable<T> = {
   toJSON(message: T): unknown;
 };
 
+export const MAX_PAGE_SIZE = 1000; // Global maximum limit
```
[the remark this answers is not in the document]
Comment (zachterrell57) [reply apps/hubble/src/rpc/httpServer.ts:59]: @adityapk00 @manan19 cited that 5k was causing performance issues: https://github.com/farcasterxyz/hub-monorepo/issues/1715

So I just picked 1000 arbitrarily

[the remark this answers is not in the document]
Comment (sanjayprabhu) [reply apps/hubble/src/rpc/httpServer.ts:59]: We should probably call this `DEFAULT_PAGE_SIZE`? Apart from that 1K sounds fine to me for http, the serialization cost is real.

**Comment by sds** [comment]
Thanks @zachterrell57! Took your changes and added a changeset to merge them in [here](https://github.com/farcasterxyz/hub-monorepo/pull/1776). Closing in favor of that.

**Pull request closed without merging**